lib/ansible/plugins/action/__init__.py

```diff
@@ -568,14 +568,17 @@ def _remote_expand_user(self, path, sudoable=True, pathsep=None):
         expand_path = split_path[0]
 
         if sudoable and expand_path == '~' and self._play_context.become and self._play_context.become_user:
-            expand_path = '~%s' % self._play_context.become_user
+            expand_path = '~%s/' % self._play_context.become_user
+        else:
+            # use remote user instead, if none set default to current user
+            expand_path = '~%s/' % self._play_context.remote_user or self._connection.default_user or ''
```

Comment:
If the connection subclass doesn't override self.default_user and there is no _play_context.remote_user could this end up setting expand_path = '~None/' ?

Kind of a Connection subclass impl bug at that point if so...  shell.expand_user() will cause an error on ~None I assume (unless /home/None exists..?)

Reply:
if None, it will be false, that is why `or ''` will take care of that

Reply:
~None/ is used here in Ansible 2.5.3 and other people saw this, too: #40483